### PR TITLE
OH2-257 | Revise HTTP response codes in API

### PR DESCRIPTION
**Pull request by ArnaudFonzam**
See OH2-257.

### CODE REVIEW

File: src/main/java/org/isf/accounting/rest/BillController.java [hunk 1]
```diff
@@ -123,13 +123,13 @@ public ResponseEntity<FullBillDTO> newBill(@RequestBody FullBillDTO newBillDto)
 		if (pat != null) {
 			bill.setBillPatient(pat);
 		} else {
-			throw new OHAPIException(new OHExceptionMessage("Patient not found."));
+			return ResponseEntity.badRequest().body(new OHExceptionMessage("Patient not found."));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/rest/BillController.java:126]: I find three different return values for something not found:

<pre>
return ResponseEntity.badRequest().body("XXXX not found.");
return ResponseEntity.badRequest().body(new OHExceptionMessage("XXXX not found."));
return ResponseEntity.notFound().build();
</pre>

Shouldn't there only be one form?

File: src/main/java/org/isf/admtype/rest/AdmissionTypeController.java [hunk 2]
```diff
@@ -73,12 +72,12 @@ public AdmissionTypeController(AdmissionTypeBrowserManager admtManager, Admissio
...
 	@PostMapping(value = "/admissiontypes", produces = MediaType.APPLICATION_JSON_VALUE)
-	ResponseEntity<AdmissionTypeDTO> newAdmissionType(@RequestBody AdmissionTypeDTO admissionTypeDTO) throws OHServiceException {
+	ResponseEntity<?> newAdmissionType(@RequestBody AdmissionTypeDTO admissionTypeDTO) throws OHServiceException {
 		String code = admissionTypeDTO.getCode();
 		LOGGER.info("Create Admission Type {}", code);
 		AdmissionType newAdmissionType = admtManager.newAdmissionType(mapper.map2Model(admissionTypeDTO));
-		if (!admtManager.isCodePresent(code)) {
-			throw new OHAPIException(new OHExceptionMessage("Admission Type is not created."), HttpStatus.INTERNAL_SERVER_ERROR);
+		if (admtManager.isCodePresent(code)) {
+			return ResponseEntity.badRequest().body(new OHExceptionMessage("The code of Admission Type is already used."));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/admtype/rest/AdmissionTypeController.java:80]: There appears to be two different returns for something already present/being used:
<pre>
return ResponseEntity.badRequest().body(new OHExceptionMessage("XXXXX is already used."));
return ResponseEntity.internalServerError().body(new OHExceptionMessage("XXXXX is already present."));
</pre>
Shouldn't there only be one?

**Comment by SilverD3** [comment]
This PR is outdated. We should close it.
#482 addresses the issue this PR tries to solve.

**Comment by mwithi** [comment]
Superseded by https://github.com/informatici/openhospital-api/pull/482

**Pull request closed without merging**